src/widget/wcuemenupopup.cpp

```diff
@@ -40,6 +53,28 @@ WCueMenuPopup::WCueMenuPopup(UserSettingsPointer pConfig, QWidget* parent)
     m_pDeleteCue->setObjectName("CueDeleteButton");
     connect(m_pDeleteCue, &QPushButton::clicked, this, &WCueMenuPopup::slotDeleteCue);
 
+    m_pSavedLoopCue = new CueTypePushButton(this);
+    m_pSavedLoopCue->setToolTip(
+            tr("Toggle this cue type between normal cue and saved loop, using "
+               "the current beatloop size or the current play position") +
+            "\n\n" +
+            tr("Left-click: Toggle between normal cue and saved loop, using"
+               "the current beatloop size as the loop size") +
+            "\n" +
+            tr("Right-click: If the current play position is after the cue,"
+               "set that position as loop end and make the cue a saved loop "
+               "if not"));
```

Comment:
```suggestion
            tr("Toggle this cue type between normal cue and saved loop")
            "\n\n" +
            tr("Left-click: Use the old size or the current beatloop size as the loop size") +
            "\n" +
            tr("Right-click: Use the current play position as loop end if it is after the cue);
```

Reply:
I am not sure if keeping the old loop size is the intended behaviour, because it is invisible, when the cue is an normal cue. We loose the "feature" to adjust the beatloops size of a saved loop by this.

Reply:
This was [requested](https://github.com/mixxxdj/mixxx/pull/13215#issuecomment-2098297092) by @Swiftb0y
Since you can change the loop size with right-click, I guess it is okay.
Perhaps I could also make double-click to change the loop size with the beatloops size value?

Reply:
> Perhaps I could also make double-click

I suggest to keep it simple.
The purpose of this button is to quickly change the type, with two loop options.
Adjusting saved loops should better be done with the looping controls IMO.

Adjusting the right-click behaviour might work:
* if before cue: reject
* if at cue: use beatloop size **NEW**
* if after cue: use play pos

Reply:
That should be the behaviour. Also with your commit, the end pos is now removed after popup close so the left button may use the beatloop size again on next attempt

Reply:
> I suggest to keep it simple.

I also thinks that important. if the functionality for retaining the previous size results in code that is too complicated, we should ensure the code remains understandable. In the worst case, the user accidentally clicks the wrong button (that is not easy to access in the first place) and they learn that it changes the length the hard way.

I'm sorry for being so indecisive on this.

Reply:
No problem. Could you please have a look with the current solution? It will allow to quickly revert the saved loop, and will "clean" the dangling end position when the popup is closed, thanks to @ronso0's work. I think the code remain quite simple, so hopefully we have a good trade-off between usability and code simplicity.